src/data_source/storage/sql.rs

```diff
@@ -1787,6 +1789,636 @@ impl SqlStorage {
+                            message: format!("failed to get column size {e}"),
+                        })?;
+                        let num_transactions: i64 =
+                            row.try_get("transactions").map_err(|e| QueryError::Error {
+                                message: format!("failed to get column transactions {e}"),
+                            })?;
+
+                        let height: u64 = height.try_into().map_err(|e| QueryError::Error {
+                            message: format!("failed to convert height to u64 {e}"),
+                        })?;
+                        let timestamp: u64 =
+                            timestamp.try_into().map_err(|e| QueryError::Error {
+                                message: format!("failed to convert timestamp to u64 {e}"),
+                            })?;
+                        let time: u64 = time.unwrap_or(0).try_into().map_err(|e| QueryError::Error {
+                            message: format!("failed to convert time to u64 {e}"),
+                        })?;
+                        let size: u64 = size.try_into().map_err(|e| QueryError::Error {
+                            message: format!("failed to convert size to u64 {e}"),
+                        })?;
+                        let num_transactions: u64 =
+                            num_transactions.try_into().map_err(|e| QueryError::Error {
+                                message: format!("failed to convert num_transactions to u64 {e}"),
+                            })?;
+
+                        Ok((height, timestamp, time, size, num_transactions))
+                    })
+                })
+                .try_fold(
+                    explorer::data_source::ExplorerHistograms {
+                        block_time: Vec::with_capacity(50),
+                        block_size: Vec::with_capacity(50),
+                        block_transactions: Vec::with_capacity(50),
+                        block_heights: Vec::with_capacity(50),
+                    },
+                    |mut histograms: explorer::data_source::ExplorerHistograms,
+                     row: Result<(u64, u64, u64, u64, u64), QueryError>| async {
+                        let (height, _timestamp, time, size, num_transactions) = row?;
+                        histograms
+                            .block_time
+                            .push(time);
+                        histograms.block_size.push(size);
+                        histograms.block_transactions.push(num_transactions);
+                        histograms.block_heights.push(height);
+                        Ok(histograms)
+                    },
+                ).await;
+
+            histograms
+        }.await?;
+
+        let genesis_overview = async {
+            let row = self
+                .query_one(
+                    "SELECT
+                    (SELECT COUNT(*) FROM header) AS blocks,
```

Comment:
Better way to get number of blocks since genesis is `SELECT max(height) FROM header`: works even if some blocks are missing and is much faster

Reply:
This is a very fair!

Reply:
This has been adjusted in commit `f5486981d52d5826f9de5f4bd7bc30448e79abc0`.